src/main/java/de/tum/cit/aet/artemis/communication/service/ReactionService.java

```diff
@@ -195,4 +214,23 @@ private Reaction createReactionForPost(Reaction reaction, Post posting, User use
         api.broadcastForPost(new PostDTO(post, MetisCrudAction.UPDATE), course.getId(), null);
         return savedReaction;
     }
+
+    private Course getReactionCourse(Reaction reaction) {
+        if (reaction.getPost() != null) {
+            return reaction.getPost().getCoursePostingBelongsTo();
+        }
+        if (reaction.getAnswerPost() != null) {
+            return reaction.getAnswerPost().getCoursePostingBelongsTo();
+        }
+        return null;
+    }
```

Comment:
Considerung how we currently use the method I think it should throw an error instead in the null case